src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -1014,5 +1029,119 @@ private static bool ProjectHasPackagesConfigFile(string projectDirectory, string
 
             return false;
         }
+
+        public static async Task<bool> RestoreAsync(
```

Comment:
I had to make this public because `NuGet.Build.Tasks` compiles in shared classes.  Should I make `InternalsVisibleTo` and see if I can disable the shared compilation classes?